RealmSwift/List.swift

```diff
@@ -526,7 +538,7 @@ extension List: RealmCollection {
 }
 
 // MARK: - MutableCollection conformance, range replaceable collection emulation
-extension List: MutableCollection {
+extension List: MutableCollection, RangeReplaceableCollection {
```

Comment:
List deliberately doesn't conform to RangeReplaceableCollection because not all of the operations it requires make sense for List.